### PR TITLE
created new form whit validations

### PR DESCRIPTION
**Pull request by leonardofrp5**
**Summary**
I create new form whit angular and added some validations 

**Screenshots**
Screen Shot 2019-11-27 at 10 41 04

**Tellor Issue**
https://trello.com/c/Z48lE2zQ/8-register-layout-exercise

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,35 @@
...
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent {
-  title = 'wolox-books';
+  rForm: FormGroup;
```
Comment (fmontoy) [inline src/app/app.component.ts]: Put a more explicit name eg. Register form

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,10 +1,35 @@
 import { Component } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss']
 })
 export class AppComponent {
```
Comment (fmontoy) [inline src/app/app.component.ts]: Create a new component for this, don't use the AppComponent!

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,10 +1,35 @@
...
+  rForm: FormGroup;
+  post: any;
+  firstName: string = '';
+  lastName: string = '';
+  email: string = '';
+  passwordConfirmation: string = '';
+  messageAlert: string = 'Campo obligatorio'
+
+  constructor(private fb: FormBuilder){
+    this.rForm = fb.group({
```
Comment (fmontoy) [inline src/app/app.component.ts]: No need to declare the form in the constructor, you can create it as a class attribute

File: src/app/app.component.scss [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  padding: 10px;
+}
+
+.section-input{
+  display: flex;
+  flex-direction: column;
+  height: 80px;
+  padding: 5px 25px;
+
+  label{
```
Comment (fmontoy) [inline src/app/app.component.scss]: avoid using html tags as selectors, use classes instead :)

File: src/app/app.component.scss [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    display: flex;
+    font-family: 'Roboto', sans-serif;
+    size: $ContentText;
+    padding-bottom: 5px;
+    padding-top: 5px;
+    color: $Black;
+  }
+
+  input{
```
Comment (fmontoy) [inline src/app/app.component.scss]: avoid using html tags as selectors, use classes instead :)

File: src/app/app.component.ts (hunk 2 above)
Comment (fmontoy) [reply src/app/app.component.ts]: AppComponent's HTML should only have the <router-outlet></router-outlet> tag, create a new component (with the cli `ng generate component screens/component-name`), put all the html and ts that you already created in this component :)

File: src/app/form-register/form-register.component.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-form-register',
+  templateUrl: './form-register.component.html',
+  styleUrls: ['./form-register.component.scss']
+})
+export class FormRegisterComponent implements OnInit {
```
Comment (fmontoy) [inline src/app/form-register/form-register.component.ts]: put the form-register component folder in a `app/screens`folder :)

File: src/app/form-register/form-register.component.html [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    <form [formGroup]='registerForm' (ngSubmit)='addPost(registerForm.value)'>
+      <div class="section-input">
+        <label class="text-label">Nombre</label>
+        <input class="decoration-input" type="text" formControlName="firstName" />
+        <span class="message-alert" *ngIf="!registerForm.controls['firstName'].valid && registerForm.controls['firstName'].touched">{{ messageAlert }}</span>
+      </div>
+      <div class="section-input">
+        <label class="text-label">Apellido</label>
+        <input class="decoration-input" type="text" formControlName="lastName" />
+        <span class="message-alert" *ngIf="!registerForm.controls['lastName'].valid && registerForm.controls['lastName'].touched">{{ messageAlert }}</span>
```
Comment (fmontoy) [inline src/app/form-register/form-register.component.html]: don't use `registerForm.controls['firstName'].valid` use `registerForm.controls.firstName.valid` instead

File: src/app/form-register/form-register.component.html [hunk 8]
```diff
@@ -0,0 +1,30 @@
+<div class="from-container" *ngIf="!name; else forminfo">
+    <img class="logo-wolox" alt="logo-wolox" src="../../assets/LogoWolox-Original.png">
```
Comment (fmontoy) [inline src/app/form-register/form-register.component.html]: Use relative routes `assets/img.png`

File: src/app/form-register/form-register.component.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+  addPost(post) {
+    this.firstName = post.firstName;
+    this.lastName = post.lastName;
+    this.email = post.email;
+    this.passwordConfirmation = post.passwordConfirmation;
+    console.log(post);
+  }
+
+  ngOnInit() {
```
Comment (fmontoy) [inline src/app/form-register/form-register.component.ts]: I always try to put ngOnInit after the constructor.

File: src/app/form-register/form-register.component.html [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+        <input class="decoration-input" type="password" formControlName="passwordConfirmation" />
+        <span class="message-alert" *ngIf="!registerForm.controls['passwordConfirmation'].valid && registerForm.controls['passwordConfirmation'].touched">{{ messageAlert }}</span>
+      </div>
+      <div class="section-buttons">
+        <input class="button button-singUp" type="submit" value="Sing up"/>
+        <input class="button button-login" type="submit" [disabled]="!registerForm.valid" value="Login"/>
+      </div>
+    </form>
+  </div>
+
```
Comment (fmontoy) [inline src/app/form-register/form-register.component.html]: No new line :)

File: src/app/form-register/form-register.component.ts (hunk 9 above)
Comment (leonardofrp5) [reply src/app/form-register/form-register.component.ts]: this is a  lifecycle ? @fmontoy

File: src/app/screens/form-register/form-register.component.scss [hunk 11]
```diff
@@ -0,0 +1,81 @@
+@import '../../../scss/variables/colors.scss';
+@import '../../../scss/variables/fonts.scss';
```
[the remark this answers is not in the document]
Comment (fmontoy) [reply src/app/screens/form-register/form-register.component.scss]: .

File: src/app/screens/form-register/form-register.component.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  styleUrls: ['./form-register.component.scss']
+})
+export class FormRegisterComponent implements OnInit {
+
+  registerForm: FormGroup;
+  post: any;
+  firstName: string = '';
+  lastName: string = '';
+  email: string = '';
+  passwordConfirmation: string = '';
```
[the remark this answers is not in the document]
Comment (fmontoy) [reply src/app/screens/form-register/form-register.component.ts]: .

File: src/app/screens/form-register/form-register.component.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  private buildForm(){
+    this.registerForm = this.fb.group({
+      'firstName': [null, Validators.compose([Validators.required, Validators.minLength(3)])],
+      'lastName': [null, Validators.compose([Validators.required, Validators.minLength(3)])],
+      'email': [null, Validators.compose([Validators.required, Validators.email])],
+      'passwordConfirmation': [null, Validators.compose([Validators.required, Validators.maxLength(20), Validators.minLength(5)])]
+    });
+  }
+
+  addPost(post) {
```
[the remark this answers is not in the document]
Comment (fmontoy) [reply src/app/screens/form-register/form-register.component.ts]: .

File: src/app/screens/form-register/form-register.component.ts [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+    this.firstName = post.firstName;
```
[the remark this answers is not in the document]
Comment (fmontoy) [reply src/app/screens/form-register/form-register.component.ts]: .

**Pull request merged**